### ISSUE TITLE
Zero-shot Performance of InstructBLIP on OK-VQA Dataset

### ISSUE THREAD

**Issue by Upper9527** [salesforce/LAVIS#534]
Dear authors,

Thanks for the great work! I wonder to know the zero-shot performance of InstructBLIP on OK-VQA Dataset. However, it's not report in the paper. I reproduced this and got 57.6%. Can you tell me the results you got?

**Comment by NaiquanChai** [comment]
Hi. Can you share how reproduced? I only got 17% accuracy on OK-VQA Dataset.

**Comment by Fym68** [comment]
@NaiquanChai Could you provide evaluation script or implementation guideline for okvqa？Sincere thanks！

**Comment by gabrielsantosrv** [comment]
@Upper9527 could you share your evaluation script? I got only 42.38% :(
I suppose I'm passing some wrong parameters to the `generate` method. Specifically, I'm using `max_length=10` and `length_penalty=-1`

Here is my evaluation script.
 ```
   args = parse_args()
    config = OmegaConf.load(args.setup)
    print(config)

    print(">>>>> Load model")
    device = torch.device(f"cuda:{args.gpu}" if torch.cuda.is_available() else "cpu")
    model, vis_processors, txt_processors = load_model_and_preprocess(
        name="blip2_vicuna_instruct", model_type="vicuna7b", is_eval=True, device=device
    )
    model.llm_model.config.pad_token_id = 2

    print(model)
    model = model.to(device)
    model.eval()

    def processor(images, text):
        input_images = torch.stack([vis_processors["eval"](img) for img in images], dim=0)
        return {"image": input_images, "prompt": text}

    dataset = load_vqa_dataset(config=config, processor=processor)

    results = []
    for i, batch in tqdm.tqdm(enumerate(dataset), desc="Inference"):
        question_ids, inputs, answers = batch
        inputs["image"] = inputs["image"].to(device)

        generated_texts = model.generate(samples=inputs, max_length=10, length_penalty=-1)

        for question_id, gen_answer in zip(question_ids, generated_texts):
            results.append({"question_id": question_id, "answer": gen_answer.strip()})

    with open(args.output_file, "w") as f:
        json.dump(results, f)
```
